DataTrackExample/ViewController.swift

```diff
@@ -0,0 +1,468 @@
+            drawer.shapeLayer.path = combinedPath
+        } else {
+            drawer.shapeLayer.path = path.cgPath
+        }
+        
+        if (self.view.layer.sublayers?.contains(drawer.shapeLayer!) == false) {
+            drawer.shapeLayer.strokeColor = drawer.color
+            self.view.layer.addSublayer(drawer.shapeLayer!)
+            self.view.bringSubview(toFront: disconnectButton)
+        }
+
+    }
+    
+    func addDrawer(_ key: TVIDataTrack, color: CGColor) {
+        let drawer = Drawer(color)
+        drawers[key] = drawer
+    }
+    
+    func removeDrawer(_ key: TVIDataTrack) {
+        if let drawer = drawers[key] {
+            drawer.shapeLayer.removeFromSuperlayer()
+            drawers.removeValue(forKey: key)
+        }
+    }
+}
+
+// MARK: UITextFieldDelegate
+extension ViewController : UITextFieldDelegate {
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        self.connect(sender: textField)
+        return true
+    }
+}
+
+// MARK: TVIRoomDelegate
+extension ViewController : TVIRoomDelegate {
+    func didConnect(to room: TVIRoom) {
+        
+        self.localParticipant = room.localParticipant!
+        self.addDrawer(self.localDataTrack, color: UIColor.black.cgColor)
+        
+        if room.remoteParticipants.count > 0 {
+            room.remoteParticipants[0].delegate = self
+        }
+
+        logMessage(messageText: "Connected to room \(room.name) as \(String(describing: room.localParticipant?.identity))")
+    }
+    
+    func room(_ room: TVIRoom, didDisconnectWithError error: Error?) {
+        logMessage(messageText: "Disconncted from room \(room.name), error = \(String(describing: error))")
+        
+        for (_, drawer) in self.drawers {
+            drawer.shapeLayer!.removeFromSuperlayer()
+        }
+        self.drawers.removeAll(keepingCapacity: true)
+        self.localParticipant = nil;
+        self.room = nil
+        
+        self.showRoomUI(inRoom: false)
+    }
+    
+    func room(_ room: TVIRoom, didFailToConnectWithError error: Error) {
+        logMessage(messageText: "Failed to connect to room with error: \(error.localizedDescription)")
+        self.room = nil
+        
+        self.showRoomUI(inRoom: false)
+    }
+    
+    func room(_ room: TVIRoom, participantDidConnect participant: TVIRemoteParticipant) {
+        participant.delegate = self
+        
+        logMessage(messageText: "Participant \(participant.identity) connected with \(participant.remoteAudioTracks.count) audio and \(participant.remoteVideoTracks.count) video tracks")
+    }
+    
+    func room(_ room: TVIRoom, participantDidDisconnect participant: TVIRemoteParticipant) {
+        logMessage(messageText: "Room \(room.name), Participant \(participant.identity) disconnected")
+    }
+}
+
+// MARK: TVIParticipantDelegate
+extension ViewController : TVIRemoteParticipantDelegate {
+    func remoteParticipant(_ participant: TVIRemoteParticipant,
```

Comment:
Do we want to provide implementations for the video and audio related delegates? We aren't displaying video or providing audio so this just seems to be inflating the size of the quickstart.

Reply:
That's the question I had too. I don't see any value but I had kept them here for the discussion. should I remove them?

Reply:
I think it would be OK to remove them. Lets just show the things that we need to make data tracks work.